tool/update_version.sh

```diff
@@ -29,17 +29,17 @@ pushd packages
 # We could use LAST_VERSION instead of allowing any previous version
 
 # Update the version of all packages.
-perl -pi -e "s/^(\\W*version:) [0-9.]+/\$1 $VERSION/g" $PUBSPECS
+perl -pi -e "s/^(\\W*version:) [0-9.dev-]+/\$1 $VERSION/g" $PUBSPECS
 
 # Update all references to package versions
-perl -pi -e "s/^(\\W*devtools_shared:) \\^?[0-9.]+/\$1 $VERSION/g" $PUBSPECS
-perl -pi -e "s/^(\\W*devtools_server:) \\^?[0-9.]+/\$1 $VERSION/g" $PUBSPECS
-perl -pi -e "s/^(\\W*devtools:) \\^?[0-9.]+/\$1 $VERSION/g" $PUBSPECS
-perl -pi -e "s/^(\\W*devtools_app:) \\^?[0-9.]+/\$1 $VERSION/g" $PUBSPECS
-perl -pi -e "s/^(\\W*devtools_testing:) \\^?[0-9.]+/\$1 $VERSION/g" $PUBSPECS
+perl -pi -e "s/^(\\W*devtools_shared:) \\^?[0-9.dev-]+/\$1 $VERSION/g" $PUBSPECS
```

Comment:
As a drive by comment, consider moving the functionality in `update_version.sh`  to the dart cli tool in `//tool/`.

Reply:
yeah that would be nice particularly if it was done using a real yaml parser to make the changes more robust. For completely quick hacks I do like having them as shell scripts were there is no ambiguity that it is just a quick hack.